### PR TITLE
Add CMakeLists.txt

### PR DESCRIPTION
**Pull request by encounter**
Providing a CMake project makes it easier to get started in IDEs (like CLion).

Example release build with ninja generator:
```
cmake -B build -GNinja -DCMAKE_BUILD_TYPE=RelWithDebInfo
cmake --build build
```

No worries if you prefer just having a Makefile, though I do personally prefer CMake over a manual Makefile in general.

### CODE REVIEW

**Comment by ethteck** [comment]
I could go either way on this. I'm not as familiar with CMake, but I know it's a bit smarter than plain Makefiles. However, if we are going to do this, I think we should fully move to it and not end up supporting two different ways of building the program. If we go this route, the github actions stuff should move to this as well.

**Comment by encounter** [comment]
Cool! I updated the PR to use CMake in CI as well.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,26 +10,30 @@ jobs:
       - name: Install GCC multilib
         run: |
           sudo apt-get update
-          sudo apt-get install -y gcc-multilib g++-multilib
+          sudo apt-get install -y gcc-multilib g++-multilib ninja-build
```
Comment (Treeki) [inline .github/workflows/ci.yml:13]: Should this be installing cmake as well?

Comment (encounter) [reply .github/workflows/ci.yml:13]: cmake is a preinstalled on the GH actions runner images, and they provide a newer version than you’d get from apt.

**Comment by ethteck** [comment]
Looks good! The only other things are that we should remove the Makefile, and the readme should be updated to explain how to build the project (for debugging, as well as maybe for release).

**Comment by encounter** [comment]
README updated for CMake build instructions & Makefile removed.

**Pull request merged**